scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -711,7 +711,7 @@ def parse_arguments(parser, args, options = None):
         options.enable_coverage = True
 
     if not options.coverage_platform:
-        options.coverage_platform = options.platform
+        options.coverage_platform = options.platform or []
```

Comment:
do you think the better solution is to set `--platform` option default to [] ?
it will also fix another issue possible here at line 720